### PR TITLE
[15.0][mig]account_payment_term_discount

### PR DESCRIPTION
**Pull request by leemannd**
The previous migration was never merged: https://github.com/OCA/account-payment/pull/537
I have taken it with the following modifications:
- squashing of administrative commits
- squashing of migration commits for the version 15.0

### CODE REVIEW

**Comment by leemannd** [comment]
Hello @cshardey I have updated your migration and created a new PR. Your review is welcome

**Review by cyrilmanuel — APPROVED** [review]

**Comment by leemannd** [comment]
Hello @max3903 we are migrating this module from OpensourceIntegrators. Do you have time to have a look at it and eventually drop a review?

**Comment by leemannd** [comment]
Hello is any member of @OCA/accounting-maintainers Available to make a review and eventually merge this PR?

File: account_payment_term_discount/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2010-2012 Camptocamp Austria (<http://www.camptocamp.at>)
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Account Payment Terms Discount",
+    "version": "15.0.1.1.5",
```
Comment (max3903) [inline account_payment_term_discount/__manifest__.py]:
```suggestion
    "version": "15.0.1.0.0",
```

File: account_payment_term_discount/i18n/account_payment_term_discount.pot [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+#. module: account_payment_term_discount
+#: model:ir.model,name:account_payment_term_discount.model_account_payment_register
+msgid "Register Payment"
+msgstr ""
+
+#. module: account_payment_term_discount
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_bank_statement_line__shipping_lines_total
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_move__shipping_lines_total
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_payment__shipping_lines_total
+msgid "Shippling Lines Total"
```
Comment (max3903) [inline account_payment_term_discount/i18n/account_payment_term_discount.pot]:
```suggestion
msgid "Shipping Lines Total"
```

File: account_payment_term_discount/i18n/de.po [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+#. module: account_payment_term_discount
+#: model:ir.model,name:account_payment_term_discount.model_account_payment_register
+msgid "Register Payment"
+msgstr ""
+
+#. module: account_payment_term_discount
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_bank_statement_line__shipping_lines_total
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_move__shipping_lines_total
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_payment__shipping_lines_total
+msgid "Shippling Lines Total"
```
Comment (max3903) [inline account_payment_term_discount/i18n/de.po]:
```suggestion
msgid "Shipping Lines Total"
```

File: account_payment_term_discount/i18n/es_AR.po [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+#. module: account_payment_term_discount
+#: model:ir.model,name:account_payment_term_discount.model_account_payment_register
+msgid "Register Payment"
+msgstr "Registrar Pago"
+
+#. module: account_payment_term_discount
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_bank_statement_line__shipping_lines_total
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_move__shipping_lines_total
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_payment__shipping_lines_total
+msgid "Shippling Lines Total"
```
Comment (max3903) [inline account_payment_term_discount/i18n/es_AR.po]:
```suggestion
msgid "Shipping Lines Total"
```

File: account_payment_term_discount/i18n/es_AR.po [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+#. module: account_payment_term_discount
+#: code:addons/account_payment_term_discount/wizard/account_payment_register.py:0
+#, python-format
+msgid ""
+"This method should only be called to process a single invoice's payment."
+msgstr ""
+"Este método solo debe invocarse para procesar el pago de una única factura."
+
+#~ msgid "Exclude Shippling Amount"
+#~ msgstr "Excluir Monto del Envío"
```
Comment (max3903) [inline account_payment_term_discount/i18n/es_AR.po]:
```suggestion
```

File: account_payment_term_discount/i18n/fr_FR.po [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+#. module: account_payment_term_discount
+#: model:ir.model,name:account_payment_term_discount.model_account_payment_register
+msgid "Register Payment"
+msgstr "Enregistrer le paiement"
+
+#. module: account_payment_term_discount
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_bank_statement_line__shipping_lines_total
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_move__shipping_lines_total
+#: model:ir.model.fields,field_description:account_payment_term_discount.field_account_payment__shipping_lines_total
+msgid "Shippling Lines Total"
```
Comment (max3903) [inline account_payment_term_discount/i18n/fr_FR.po]:
```suggestion
msgid "Shipping Lines Total"
```

File: account_payment_term_discount/i18n/fr_FR.po [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+#: code:addons/account_payment_term_discount/wizard/account_payment_register.py:0
+#, python-format
+msgid ""
+"This method should only be called to process a single invoice's payment."
+msgstr ""
+"Cette méthode ne doit être appelée que pour traiter le paiement d'une seule "
+"facture."
+
+#~ msgid "Exclude Shippling Amount"
+#~ msgstr "Exclure le montant des frais de port"
```
Comment (max3903) [inline account_payment_term_discount/i18n/fr_FR.po]:
```suggestion
```

File: account_payment_term_discount/models/account_move.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        compute="_compute_discount_date",
+        help="Compute discount on the invoice based on the payment term discount "
+        "percentage."
+        "and the current date",
+        store=True,
+    )
+    discount_taken = fields.Monetary(copy=False)
+    check_payment_discount = fields.Boolean(compute="_compute_payment_disc")
+    shipping_lines_total = fields.Monetary(
+        "Shippling Lines Total",
```
Comment (max3903) [inline account_payment_term_discount/models/account_move.py]:
```suggestion
        "Shipping Lines Total",
```

**Review by max3903 — CHANGES_REQUESTED** [review]

**Comment by leemannd** [comment]
Hello @max3903 , thank you for the review. I have updated the PR accordingly to your review.

**Comment by max3903** [comment]
@leemannd You only resolved my comments but there is no new commits with my suggested changes.

**Comment by leemannd** [comment]
@max3903 I forgot to push. This is done now

**Comment by max3903** [comment]
@leemannd Can you add tests and squash the migration commits together?

**Review by max3903 — APPROVED** [review]

**Comment by leemannd** [comment]
Hello, I have squashed the commits together. I'm running a bit out of time to update the tests. Feel free to make a PR a aginst my branch and I'll merge them.

**Comment by leemannd** [comment]
Hello @max3903 Can you reopen it and merge it?
I'm currently not considering adding unit tests:
- the coverage is "good enough"
- AFAIR this is not a a blocking point for the migration of a module
- your company migrated the module into the previous version and never bothered improving the coverage

**Comment by leemannd** [comment]
Closing the PR as the maintainer of the module is not willing to merge it

**Pull request closed without merging**